graphix/linalg_validations.py

```diff
@@ -2,29 +2,41 @@
 
 from __future__ import annotations
 
+import warnings
+from typing import TypeVar
+
 import numpy as np
+import numpy.typing as npt
+
+_T = TypeVar("_T", bound=np.generic)
+
+
+def is_square(matrix: npt.NDArray[_T]) -> bool:
+    """Check if matrix is square."""
+    if matrix.ndim != 2:
+        return False
+    rows, cols = matrix.shape
+    return rows == cols
 
 
-def check_square(matrix: np.ndarray) -> bool:
+def is_qubitop(matrix: npt.NDArray[_T]) -> bool:
     """Check if matrix is a square matrix with a power of 2 dimension."""
-    if len(matrix.shape) != 2:
-        raise ValueError(f"The object has {len(matrix.shape)} axes but must have 2 to be a matrix.")
-    if matrix.shape[0] != matrix.shape[1]:
-        raise ValueError(f"Matrix must be square but has different dimensions {matrix.shape}.")
-    size = matrix.shape[0]
-    if size & (size - 1) != 0:
-        raise ValueError(f"Matrix size must be a power of two but is {size}.")
-    return True
+    if not is_square(matrix):
+        warnings.warn(f"is_square is not called before {is_qubitop.__name__}.", stacklevel=1)
```

Comment:
Why such a warning? Why should I write `is_square(m) and is_qubit(m)` instead of just `is_qubit(m)` (that would call `is_square` internally in any case)? The doc-comment suggests that `is_qubit(m)` should really be equivalent to `is_square(m) and is_qubit(m)`.

Reply:
It's originating from the original implementation.
Let me refactor it!

Reply:
Fixed in 65eb69a2c5db92b2b77acb19c4f55f2f27d97a56 .